experiments/AMIP/modular/components/ocean/eisenman_seaice.jl

```diff
@@ -0,0 +1,293 @@
+    ρ::FT = 1500                    # sea ice density [kg / m3]
+    c::FT = 800                     # sea ice heat capacity [J / kg / K]
+    z0m::FT = 1e-3                  # roughness length for momentum [m]
+    z0b::FT = 1e-5                  # roughness length for buoyancy [m]
+    C0_base::FT = 120               # ice base transfer coefficient [W / m2 / K]
+    T_base::FT = 273.16             # ice base temperature [K]
+    L_ice::FT = 3e8                 # latent heat coefficient for ice [J / m3]
+    T_freeze::FT = 273.16           # temperature at freezing point [K]
+    k_ice::FT = 2                   # thermal conductivity of ice [W / m / K]
+    α::FT = 0.38                    # albedo
+    σ::FT = 5.67e-8                 # Stefan Boltzmann constant [W / m^2 / K^4]
+end
+
+Base.@kwdef struct EisenmanOceanParameters{FT <: AbstractFloat}
+    h::FT = 1                       # mixed layer depth [m]
+    ρ::FT = 800                    # density times heat transfer coefficient for mixed layer [kg / m^3]
+    c::FT = 4000                    # mixed layer heat capacity [J / kg / K ]
+    z0m::FT = 1e-3                  # roughness length for momentum [m]
+    z0b::FT = 1e-5                  # roughness length for buoyancy [m]
+    α::FT = 0.06                    # albedo
+end
+
+"""
+    state_init(p::EisenmanIceParameters, space)
+
+Initialize the state vectors for the Eisenman-Zhang sea ice model.
+"""
+function state_init(p::EisenmanIceParameters, space::Spaces.AbstractSpace)
+    Y = Fields.FieldVector(
+        T_sfc = ones(space) .* p.T_freeze,
+        h_ice = zeros(space),
+        T_ml = ones(space) .* 277,
+        q_sfc = ClimaCore.Fields.zeros(space),
+    )
+    Ya = Fields.FieldVector(
+        F_turb = ClimaCore.Fields.zeros(space),
+        ∂F_turb_energy∂T_sfc = ClimaCore.Fields.zeros(space),
+        F_rad = ClimaCore.Fields.zeros(space),
+        F_base = ClimaCore.Fields.zeros(space),
+        ocean_qflux = ClimaCore.Fields.zeros(space),
+        ρ_sfc = ClimaCore.Fields.zeros(space),
+    )
+    return Y, Ya
+end
+
+"""
+    get_∂F_rad_energy∂T_sfc(T_sfc, p)
+
+Calculate the derivative of the radiative flux with respect to the surface temperature.
+"""
+function get_∂F_rad_energy∂T_sfc(T_sfc, p)
+    FT = eltype(T_sfc)
+    @. FT(4) * (FT(1) - p.α) * p.σ * T_sfc^3
+end
+
+"""
+    solve_eisenman_model!(Y, Ya, p, Δt::FT)
+
+Solve the Eisenman-Zhang sea ice model for one timestep.
+"""
+function solve_eisenman_model!(Y, Ya, p, thermo_params, Δt)
+
+    # model parameter sets
+    (; p_i, p_o) = p
+
+    # prognostic
+    (; T_sfc, h_ice, T_ml, q_sfc) = Y
+
+    # auxiliary
+    (; F_turb, ∂F_turb_energy∂T_sfc, F_rad, ocean_qflux, F_base) = Ya
+
+    # local
+    F_atm = @. F_turb + F_rad
+    ∂F_atmo∂T_sfc = get_∂F_rad_energy∂T_sfc.(T_sfc, Ref(p_i)) .+ ∂F_turb_energy∂T_sfc
+
+    # ice thickness and mixed layer temperature changes due to atmosphereic and ocean fluxes
+    if parent(h_ice)[1] > 0 # ice-covered
```

Comment:
```suggestion
    ice_covered = parent(h_ice)[1] > 0
    
    if ice_covered
```

this pattern can be used throughout to make the various if-statements / conditions a little easier to read. Similarly,

```julia
transitioning_to_ice_free = (parent(h_ice)[1] > 0) & (parent(h_ice .+ Δh_ice)[1] <= 0)
```

etc

Reply:
Great idea, done!